projects/homework_103_nourish/src/main.c

```diff
@@ -0,0 +1,53 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+static void interrupt_callback_fn(const GpioAddress *address, void *context) {
+  GpioAddress *address = (GpioAddress *)context;
+  uint16_t value = 0;
+  adc_read_converted(&address, *value);
```

Comment:
I think you meant `adc_read_converted_pin`, which takes a `GpioAddress` rather than `adc_read_converted`, which takes an `AdcChannel`. You've also got this backwards, `adc_read_converted_pin` takes a `GpioAddress` and a `uint16 *` pointer, so you need to dereference `address` and take the address of `value`.

Reply:
done